backend/app/routers/datasets.py

```diff
@@ -461,6 +461,48 @@ async def save_file(
         return new_file.dict()
     raise HTTPException(status_code=404, detail=f"Dataset {dataset_id} not found")
 
+@router.post("/{dataset_id}/filesMultiple", response_model=List[FileOut])
+async def save_files(
+    dataset_id: str,
+    files: List[UploadFile],
+    folder_id: Optional[str] = None,
+    user=Depends(get_current_user),
+    fs: Minio = Depends(dependencies.get_fs),
+    es=Depends(dependencies.get_elasticsearchclient),
+    rabbitmq_client: BlockingChannel = Depends(dependencies.get_rabbitmq),
+    allow: bool = Depends(Authorization("uploader")),
+):
+    files_added = []
+    for file in files:
+        if (dataset := await DatasetDB.get(PydanticObjectId(dataset_id))) is not None:
+            if user is None:
+                raise HTTPException(
+                    status_code=401, detail=f"User not found. Session might have expired."
+                )
+
+            new_file = FileDB(name=file.filename, creator=user, dataset_id=dataset.id)
+
+            if folder_id is not None:
+                if (folder := await FolderDB.get(PydanticObjectId(folder_id))) is not None:
```

Comment:
you might consider moving this check (and the if dataset is not none check) outside the files loop so you only have to check dataset and create folder once instead of doing it repeatedly for each file